ddtrace/monkey.py

```diff
@@ -1,104 +1,19 @@
-using Pin. For that, check its documentation.
-"""
-import importlib
-import logging
-import sys
-import threading
-
-from wrapt.importer import when_imported
-
-
-log = logging.getLogger(__name__)
-
-# Default set of modules to automatically patch or not
-PATCH_MODULES = {
-    'asyncio': False,
-    'boto': True,
-    'botocore': True,
-    'bottle': True,
-    'cassandra': True,
-    'celery': True,
-    'elasticsearch': True,
-    'futures': False,  # experimental propagation
-    'grpc': True,
-    'mongoengine': True,
-    'mysql': True,
-    'mysqldb': True,
-    'pymysql': True,
-    'psycopg': True,
-    'pylibmc': True,
-    'pymemcache': True,
-    'pymongo': True,
-    'redis': True,
-    'requests': True,
-    'sqlalchemy': True,
-    'sqlite3': True,
-    'aiohttp': True,  # requires asyncio (Python 3.4+)
-    'aiopg': True,
-    'aiobotocore': True,
-    'httplib': True,
-    'vertica': True,
-    'jinja2': True,
-
-    # Even though instrumentation for web frameworks can be installed manually,
-    # instrumenting should be done idempotently
-    'django': True,
-    'falcon': True,
-    'pyramid': True,
-    'pylons': True,
-    # Ignore some web framework integrations that might be configured explicitly in code
-    "flask": False,
-}
-
-_LOCK = threading.Lock()
-_PATCHED_MODULES = set()
-
-# Modules which are patched on first use
-# DEV: These modules are patched when the user first imports them, rather than
-#      explicitly importing and patching them on application startup `ddtrace.patch_all(module=True)`
-# DEV: This ensures we do not patch a module until it is needed
-# DEV: <contrib name> => <list of module names that trigger a patch>
-_PATCH_ON_IMPORT = {
-    'celery': ('celery', ),
-    'gevent': ('gevent', ),
-    'requests': ('requests', ),
-}
-
-
-class PatchException(Exception):
-    """Wraps regular `Exception` class when patching modules"""
-    pass
-
-
-def _on_import_factory(module, raise_errors=True):
-    """Factory to create an import hook for the provided module name"""
-    def on_import(hook):
-        # Import and patch module
-        path = 'ddtrace.contrib.%s' % module
-        imported_module = importlib.import_module(path)
-        imported_module.patch()
-
-    return on_import
+from .install import patch as _patch, patch_all as _patch_all
+from .utils.deprecation import deprecated
 
 
+@deprecated(message='Use ddtrace.install.patch_all instead', version='1.0.0')
 def patch_all(**patch_modules):
     """Automatically patches all available modules.
 
     :param dict \**patch_modules: Override whether particular modules are patched or not.
 
         >>> patch_all(redis=False, cassandra=False)
     """
-    modules = PATCH_MODULES.copy()
-    modules.update(patch_modules)
+    _patch_all(**patch_modules)
 
-    patch(raise_errors=False, **modules)
 
+@deprecated(message='Use ddtrace.install.patch instead', version='1.0.0')
```

Comment:
These functions have to be deprecated because we documented them publicly. I'm not sure if anyone is actually using them (since they're expose through the root `__init__.py`).

Reply:
I don't think we can actually do this, I am sure we have a ton of customers who are using `from ddtrace import patch_all; patch_all(flask=true)`

We can change the code underneath `patch_all()` and `patch()`, but we are probably best to keep this public API.

Reply:
The api `from ddtrace import patch_all; patch_all(flask=true)` won't break. But `from ddtrace.monkey ...` will now introduce a deprecation warning.

However, I think you're right that we should continue to support it. I'll remove the deprecation warnings.

Reply:
thanks, I think my biggest argument against it, is we are moving to `ddtrace.install` just for the word `install`. You could rename `install.py` to `monkey.py` and we wouldn't have to think about backwards compatible or updating docs, or deprecation or anything.